### PR TITLE
feat:[SSCA-1037]: Added functionality for default configuration.

### PR DESCRIPTION
**Pull request by shash-harness**
## Describe your changes
In case if a config is not present , we will send the default one. The naming convention for default configs in resource folder will be in the manner of name_type.json

## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

### Rebase Instructions:
To rebase your branch onto the latest changes in the target branch, simply use the following command in a comment: `/rebase`

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/56088)

### CODE REVIEW

**Comment by shash-harness** [comment]
trigger uts

File: ssca-manager/service/src/main/resources/sbomqs_scorecard.json [hunk 1]
```diff
@@ -0,0 +1,164 @@
+{
+  "name": "sbomqs",
+  "type": "scorecard",
+  "config_info": [
+    {
+      "id": "8bf62e5d-24da-4dab-84e8-7f192c7642f3",
+      "name": "Structural",
+      "config": [
+        {
+          "key": "Structural",
```
Comment (deepharness) [inline ssca-manager/service/src/main/resources/sbomqs_scorecard.json:10]: key for category enabled should be "IsEnabled" vale can be true or false

File: ssca-manager/service/src/main/java/io/harness/ssca/services/ConfigServiceImpl.java [hunk 2]
```diff
@@ -63,8 +78,36 @@ public ConfigResponseBody getConfigByNameAndType(
...
+        log.error(String.format("Could not fetch the default config for name [%s] and type [%s]", name, type));
+      }
+      if (scorecardConfig != null) {
+        scorecardConfig.setAccountId(accountId);
+        scorecardConfig.setOrgId(orgId);
+        scorecardConfig.setProjectId(projectId);
+        ZonedDateTime currentTime = ZonedDateTime.now();
+        String formattedTime = currentTime.format(DateTimeFormatter.ISO_OFFSET_DATE_TIME);
+        scorecardConfig.setCreationOn(formattedTime);
+        scorecardConfig.setConfigId(randomAlphabetic(10));
```
Comment (deepharness) [inline ssca-manager/service/src/main/java/io/harness/ssca/services/ConfigServiceImpl.java:102]: why we are not using uuid for configId ?

File: ssca-manager/service/src/main/resources/sbomqs_scorecard.json (hunk 1 above)
Comment (shash-harness) [reply ssca-manager/service/src/main/resources/sbomqs_scorecard.json:10]: Will update this, got this from the plugin end.

**Comment by shash-harness** [comment]
trigger uts

**Comment by shash-harness** [comment]
trigger uts

**Comment by shash-harness** [comment]
trigger ut2, ut7

**Review by deepharness — CHANGES_REQUESTED** [review]
> Reviewed all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/56088)* status: 2 file(s) to review. 1 change requests. 0 of 1 approvals obtained. 

___
*[`ssca-manager/service/src/main/java/io/harness/ssca/services/ConfigServiceImpl.java` line 94 at r3](https://reviewable.io/reviews/harness/harness-core/56088#-NkTYYLKDDEBbzud8Nci:-NkTYYLKDDEBbzud8Ncj:b-fmzgt2) ([raw file](https://github.com/harness/harness-core/blob/0546f0d96d332e3651c553ed350346cc3c186649/ssca-manager/service/src/main/java/io/harness/ssca/services/ConfigServiceImpl.java#L94)):*
> ```Java
>         log.error(String.format("Could not fetch the default config for name [%s] and type [%s]", name, type));
>       }
>       if (scorecardConfig != null) {
> ```

Do we need this if ? we can simplify the flow

if(configEntity == null){
//fetch from resource
//save it in db
}

 return toConfigResponse(configEntity);

**Comment by shash-harness** [comment]
trigger uts

**Comment by shash-harness** [comment]
trigger uts

**Review by deepharness — APPROVED** [review]
> Reviewed 1 of 2 files at r2, 1 of 1 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/56088)* status: :shipit: complete! ✓

**Comment by shash-harness** [comment]
trigger ut6

**Pull request merged**